docs(auth-helpers): fix typo in nextjs.mdx (#15805)
supabase/supabase: apps/docs/pages/guides/auth/auth-helpers/nextjs.mdx

When using the Supabase client on the server, you must perform extra steps to ensure the user's auth session remains active. Since the user's session is tracked in a cookie, we need to read this cookie and update it if necessary.

> In Next.js Server Components, you can read a cookie, but you can't write back to it. Middleware, on the other hand, allows you to both read and write to cookies.

Reviewer: Thanks @gurdulicluka! Simplifying this a bit more and reverting back to `allow` (I believe we're using Middleware as plural here).
```suggestion
Next.js Server Components allow you to read a cookie but not write back to it. Middleware on the other hand allow you to both read and write to cookies.
```
